### PR TITLE
Move most ocamlPackages to cmdLiner_1_1

### PR DESCRIPTION
**Pull request by bezmuth**
###### Description of changes

Update most packages that depend on cmdliner to cmdliner_1_1. Initally I was just adding dnssec, however this required cmdliner_1_1, which conflicted with the old version of cmdliner in alcotest, requiring updates to most packages that use cmdliner.

Many packages worked fine with cmdliner_1_1, however some required updates.

ocamlPackages.mdx required an update to an unstable version, however it looks like they are prepping for the next release soon.

All the packages have been tested with nox-review.

Apologies in advance if I went about this the wrong way.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/ligo/default.nix [hunk 1]
```diff
@@ -93,5 +96,6 @@ coq.ocamlPackages.buildDunePackage rec {
     license = licenses.mit;
     platforms = [ "x86_64-linux" ];
     maintainers = with maintainers; [ ulrikstrid ];
+    broken = true;
```
Comment (gytis-ivaskevicius) [inline pkgs/development/compilers/ligo/default.nix:99]: ?

File: pkgs/development/ocaml-modules/git/mirage.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  buildInputs = [
+    git git-paf awa awa-mirage dns dns-client tls
+    tls-mirage happy-eyeballs ca-certs-nss mirage-crypto ptime x509 tcpip
+    domain-name fmt ipaddr lwt mirage-clock mirage-flow mirage-random
+    mirage-time result rresult mimic mimic-happy-eyeballs hex
+
+  ];
+
+  propagatedBuildInputs = [
+  ];
```
Comment (gytis-ivaskevicius) [inline pkgs/development/ocaml-modules/git/mirage.nix:22]:
```suggestion
```

File: pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix [hunk 3]
```diff
@@ -21,5 +21,9 @@ buildDunePackage rec {
     homepage = "https://github.com/mirage/mirage-block-ramdisk";
     license = licenses.isc;
     maintainers = with maintainers; [ ehmry ];
+    # mirage-block-ramdisk uses an incompatable version of cstruct (<6.1.0) and
+    # is only dependen on by other broken packages. No update has been pushed
+    # since 2022-07-06, so might be abandoned?
+    broken = true;
```
Comment (gytis-ivaskevicius) [inline pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix:27]: Maybe we could create an issue on their repo to fix up cstruct compatibility or package an older version into nixpkgs?

**Review by gytis-ivaskevicius — COMMENTED** [review]
it would be nice if we had some automated tool to create issues about cmdliner on other repos :D

File: pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix (hunk 3 above)
Comment (bezmuth) [reply pkgs/development/ocaml-modules/mirage-block-ramdisk/default.nix:27]: Looks like I made a mistake on the comment, the package hasn't been updated since 2020-07-06.

It might be worth adding an old version of cstruct but that would prevent other packages that rely on newer versions of cstruct from building (if I understand how dune works within nix).

File: pkgs/development/compilers/ligo/default.nix (hunk 1 above)
Comment (bezmuth) [reply pkgs/development/compilers/ligo/default.nix:99]: Apologies on this, I marked it as broken when testing other packages, I'll fix this up asap.

**Comment by cole-h** [comment]
@ofborg eval

**Comment by bezmuth** [comment]
Apologies for the messiness on my end with the merges, had some issues and figured starting again would be easier.

File: pkgs/development/compilers/ligo/default.nix [hunk 4]
```diff
@@ -3,16 +3,17 @@
 , git
 , coq
 , cacert
+, ocaml-crunch
 }:
 
 coq.ocamlPackages.buildDunePackage rec {
   pname = "ligo";
-  version = "0.36.0";
+  version = "0.47.0";
```
Comment (Et7f3) [inline pkgs/development/compilers/ligo/default.nix:11]: Can you bump ligo in another PR ?

File: pkgs/development/ocaml-modules/ca-certs-nss/default.nix [hunk 5]
```diff
@@ -8,13 +8,13 @@
...
 , astring
-, cmdliner
+, cmdliner_1_1
 , alcotest
 }:
 
 buildDunePackage rec {
   pname = "ca-certs-nss";
-  version = "3.77";
+  version = "3.80";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/ca-certs-nss/default.nix:17]: Can you move it in it's own PR + change sha256 of source

File: pkgs/development/ocaml-modules/carton/default.nix [hunk 6]
```diff
@@ -2,19 +2,19 @@
...
 , bigarray-compat, checkseum, logs, psq, fmt
 , result, rresult, fpath, base64, bos, digestif, mmap, alcotest
-, crowbar, alcotest-lwt, lwt, findlib, mirage-flow, cmdliner, hxd
+, crowbar, alcotest-lwt, lwt, findlib, mirage-flow, cmdliner_1_1, hxd
 }:
 
 buildDunePackage rec {
   pname = "carton";
-  version = "0.4.3";
+  version = "0.4.4";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/carton/default.nix:10]: Can you rebase ? this change is already done in master

File: pkgs/development/ocaml-modules/ctypes_stubs_js/default.nix [hunk 7]
```diff
@@ -8,7 +8,6 @@
 buildDunePackage rec {
   pname = "ctypes_stubs_js";
   version = "0.1";
-
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/ctypes_stubs_js/default.nix:11]: Can you re add this line so we reduce diff ?

File: pkgs/development/ocaml-modules/data-encoding/default.nix [hunk 8]
```diff
@@ -3,31 +3,35 @@
...
 , json-data-encoding
 , json-data-encoding-bson
+, either
 , alcotest
 , crowbar
 }:
 
 buildDunePackage {
   pname = "data-encoding";
-  version = "0.4.0";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/data-encoding/default.nix:15]: Can you put in another PR ?

File: pkgs/development/ocaml-modules/dns/client.nix [hunk 9]
```diff
@@ -11,10 +11,25 @@ buildDunePackage {
   pname = "dns-client";
   inherit (dns) src version;
 
-  propagatedBuildInputs = [ cstruct fmt logs dns randomconv domain-name ipaddr
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/dns/client.nix:14]: if you do reformatting prefer a tree-wide one so we can add one commit to ignore in .git-ignore-revs

File: pkgs/development/ocaml-modules/dns/stub.nix [hunk 10]
```diff
@@ -15,6 +15,7 @@ buildDunePackage {
     dns-resolver
     dns-tsig
     dns-server
+    dnssec
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/dns/stub.nix:18]: I don't think it is linked ? so it can be put in another PR

File: pkgs/development/ocaml-modules/git/cohttp.nix [hunk 11]
```diff
@@ -1,19 +0,0 @@
-{ buildDunePackage, git
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/git/cohttp.nix:1]: Why removing this ?

File: pkgs/development/ocaml-modules/git/default.nix [hunk 12]
```diff
@@ -2,20 +2,20 @@
...
 , fpath, ke, logs, lwt, ocamlgraph, uri, rresult, base64, hxd
-, result, bigstringaf, optint, mirage-flow, domain-name, emile
+, result, bigstringaf, optint, mirage-flow, domain-name, emile, bigarray-compat
 , mimic, carton, carton-lwt, carton-git, ipaddr, psq, crowbar, alcotest-lwt
 }:
 
 buildDunePackage rec {
   pname = "git";
-  version = "3.5.0";
+  version = "3.9.0";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/git/default.nix:11]: Can you put in another PR ?

File: pkgs/development/ocaml-modules/index/default.nix [hunk 13]
```diff
@@ -10,7 +10,7 @@ buildDunePackage rec {
 
   src = fetchurl {
     url = "https://github.com/mirage/index/releases/download/${version}/index-${version}.tbz";
-    sha256 = "sha256-rPwNzqkWqDak2mDTDIBqIvachY1vfOIzFmwaXjZea+4=";
+    sha256 = "1vkbbqv5w6kc2qry4z3gin2rrxi2da00rlv0vaj3da0nm770vz5c";
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/index/default.nix:13]: Hash changed without version 🤔. Can you put in another PR ?

**Review by Et7f3 — CHANGES_REQUESTED** [review]
I fear that part of your PR is already integrated in master (rebasing onto master will make changes in here disappear)

Also cmdliner is already 1.1
```
nix-repl> ocamlPackages.cmdliner.version
"1.1.1"

nix-repl> ocamlPackages.cmdliner_1_1.version
"1.1.1"
```

What you can do: for each comment (I skipped some) mentioning another PR. Open a PR with this change. After that rebase this branch. Only small set should persist. For instance this one still use cmdliner 1. So you could focus this PR in removing cmdliner 1 0 and patching all package that explicitly require cmdliner_1_0

pkgs/development/tools/ocaml/ocp-build/default.nix

**Comment by wegank** [comment]
#190469
